File: backend/spring-boot-backend/src/main/java/site/golets/pomanager/model/PomPackage.java
package site.golets.pomanager.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;
import lombok.ToString;
import lombok.experimental.Accessors;
import org.apache.maven.model.Model;

@Data
@Accessors(chain = true)
@ToString(exclude = {"model"})
public class PomPackage {

    private String name;

    private String version;

    private String gitBranch;

    private int branchStatus;

    @JsonIgnore
    private Model model;

}

File: frontend/angular-pom-manager/src/app/common/pom-package.ts
export class PomPackage {

  name: string
  version: string
  gitBranch: string
  branchStatus: number

  constructor(name: string, version: string, gitBranch: string, branchStatus: number) {
    this.name = name;
    this.version = version;
    this.gitBranch = gitBranch;
    this.branchStatus = branchStatus;
  }
}
